src/components/requestV2/requestV2.js

```diff
@@ -0,0 +1,203 @@
+import { translate } from 'react-i18next';
+import { PrimaryButtonV2 } from '../toolbox/buttons/button';
+import { InputV2, AutoresizeTextarea } from '../toolbox/inputsV2';
+import ConverterV2 from '../converterV2';
+import styles from './requestV2.css';
+
+class RequestV2 extends React.Component {
+  constructor(props) {
+    super();
+
+    this.state = {
+      showQRCode: false,
+      shareLink: `lisk://wallet/send?recipient=${props.address}`,
+      linkCopied: false,
+      fields: {
+        amount: {
+          error: false,
+          value: '',
+          feedback: '',
+        },
+        reference: {
+          error: false,
+          value: '',
+          feedback: '',
+        },
+      },
+    };
+
+    this.toggleQRCode = this.toggleQRCode.bind(this);
+    this.onCopy = this.onCopy.bind(this);
+    this.handleFieldChange = this.handleFieldChange.bind(this);
+    this.updateShareLink = this.updateShareLink.bind(this);
+  }
+
+  onCopy() {
+    clearTimeout(this.timeout);
+    setTimeout(() => this.setState({
+      linkCopied: false,
+    }), 3000);
+
+    this.setState({
+      linkCopied: true,
+    });
+  }
+
+  /* istanbul ignore next */
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
+  }
+
+  toggleQRCode() {
+    this.setState(prevState => ({
+      showQRCode: !prevState.showQRCode,
+    }));
+  }
+
+  // eslint-disable-next-line max-statements
+  handleFieldChange({ target }) {
+    const { t } = this.props;
+    const messageMaxLength = 64;
+    const error = target.name === 'amount'
+      ? /([^\d.])/g.test(target.value) || /(\.)(.*\1){1}/g.test(target.value)
+      : target.value.length > messageMaxLength;
+    let feedback = '';
+
+    if (target.name === 'amount') {
+      target.value = /^\./.test(target.value) ? `0${target.value}` : target.value;
+      feedback = error ? t('Please use only digits and dots') : feedback;
+    } else if (target.name === 'reference' && target.value.length > 0) {
+      feedback = error
+        ? t('{{length}} extra characters', { length: target.value.length - messageMaxLength })
+        : t('{{length}} out of {{total}} characters left', {
+          length: messageMaxLength - target.value.length,
+          total: messageMaxLength,
+        });
+    }
+
+    const field = {
+      [target.name]: {
+        error,
+        value: target.value,
+        feedback,
+      },
+    };
+
+    const shareLink = this.updateShareLink(field);
+
+    this.setState(prevState => ({
+      shareLink,
+      fields: {
+        ...prevState.fields,
+        ...field,
+      },
+    }));
+  }
+
+  updateShareLink(newField) {
+    const fields = {
+      ...this.state.fields,
+      ...newField,
+    };
+    const { address } = this.props;
+    return Object.keys(fields).reduce((link, fieldName) => {
+      const field = fields[fieldName];
+      return field.value !== ''
+        ? `${link}&${fieldName}=${encodeURI(field.value)}`
```

Comment:
To allow `&` in reference, it should be
```suggestion
        ? `${link}&${fieldName}=${encodeURIComponent(field.value)}`
```